### PR TITLE
fix: fixed functioning of contact us button in footer

### LINKED ISSUE #1131: Bug: contact us button is broken in footer
### Description

- When the `contact us` button in footer is clicked the website becomes blank. The button neither redirects to any page nor shows any contact form. It seems like the website hangs out and user needs to refresh to see the website live again. 

- Steps to Reproduce
1. Go to https://milanhub.org/
2. Scroll down to footer of the website
3. Click on `contact us` button

- Screen recording

https://github.com/MilanCommunity/Milan/assets/116940083/a663ff8d-6a8c-453d-a30b-083a8a70062b

## Solution 1
We can show Work in progress as shown in https://milanhub.org/shop 

## Solution 2
We can develop a contact us page and create a contact form which will open when `contact us` button is clicked.

### Screenshots

Screen shot for proposed solution 1
milan_contact

### Browser 🥦

Google Chrome

### Checklist ✅

- [X] I checked and didn't find similar [issue](https://github.com/MilanCommunity/Milan/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/MilanCommunity/Milan/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (blank for no).

### LINKED ISSUE #1141: Bug: irrelevant text in console
### Description

- Irrelevant text is shown in console at all times.

## How to reproduce
1. Go to [Milanhub](https://milanhub.org/)
2. Right click on the screen and then click on `inspect`
3. Go to `console`
4. You will see the text in console : `WTF is wrong bro`

### Proposed Solution

- The text should be removed from the console.
- The bug lies in the Milan-main\src\components\Banners\Milaninfobanner.jsx

### Screenshots

https://github.com/MilanCommunity/Milan/assets/116940083/9ab406de-fd90-4359-abad-9d2b743d0b97

### Browser 🥦

Google Chrome

### Checklist ✅

- [X] I checked and didn't find similar [issue](https://github.com/MilanCommunity/Milan/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/MilanCommunity/Milan/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (blank for no).

### PR DESCRIPTION
**Pull request by MahendraDani**
closes #1131 
closes #1141

### 👷🏻 Changes made  

1. Fixed the modal component. Modal component can now have dynamic children as per the need.
2. On clicking the `contact us` button , a modal opens where user can report a problem. The Endpoint is `/user/report`.

### 📸 Screenshots 
 

https://github.com/MilanCommunity/Milan/assets/116940083/50bbb5e4-e922-4b57-9205-7ccee6eaf357

The above recording shows three things-
1. Proper functioning of `contact us` button and `modal` both.
2. Sucessfull API call to **report problem route.**
3. Proper functioning of database, where the report is stored as expected.

### CODE REVIEW

File: src/components/Modal/Modal.jsx
Comment (tamalCodes) [inline src/components/Modal/Modal.jsx:1]: We are using two different fonts `Inter` for the headings and `Poppins` for the normal text and buttons.
Please make sure that the fonts are taken care of along with the color of the text in the modal.

**Review by tamalCodes — CHANGES_REQUESTED** [review]

File: src/components/Modal/Modal.jsx
Comment (MahendraDani) [reply src/components/Modal/Modal.jsx:1]: Would you please tell me the text color hex codes ?

Comment (tamalCodes) [reply src/components/Modal/Modal.jsx:1]: For the heading, it would be `#28183b`, and for the normal text we have it as `#000000`

Comment (MahendraDani) [reply src/components/Modal/Modal.jsx:1]: Thanks, will make the necessary changes

**Comment by MahendraDani** [comment]
## Applied changes
1. Added font-family *Inter* to headers and *Poppins* to normal text in modals.
2. Removed irrelevant text from console, fixes #1141 

## Before
Screenshot (25)

## After
Screenshot (24)

@tamalCodes If there are any other changes please let me know.

File: src/components/Modal/Modal.jsx
Comment (MahendraDani) [reply src/components/Modal/Modal.jsx:1]: @tamalCodes Made changes and pushed in https://github.com/MilanCommunity/Milan/pull/1140/commits/e118422d6932bc8a399794f118937dddf7e921c4

File: src/components/Footer/Footer.css [hunk 1]
```diff
@@ -24,18 +24,20 @@
 }
 
 .reportModalHeader > h1 {
-  font-family: Montserrat, sans-serif;
-  color: #4e4e4e;
+  font-family: "Inter";
```
Comment (tamalCodes) [inline src/components/Footer/Footer.css:27]:
```suggestion
  font-family: "Inter",sans-serif;
```

File: src/components/Footer/Footer.css [hunk 2]
```diff
@@ -24,18 +24,20 @@
...
+  color: #28183b;
   font-size: 1.5rem;
   font-weight: 600;
 }
 
 .reportModal > h2 {
-  color: #4e4e4e;
+  font-family: "Poppins";
```
Comment (tamalCodes) [inline src/components/Footer/Footer.css:34]:
```suggestion
  font-family: "Poppins"sans-serif;
```

File: src/components/Footer/Footer.css [hunk 3]
```diff
@@ -24,18 +24,20 @@
...
+  color: #28183b;
   font-size: 0.9rem;
 }
 
 #reportForm {
+  font-family: "Poppins";
```
Comment (tamalCodes) [inline src/components/Footer/Footer.css:40]:
```suggestion
  font-family: "Poppins"sans-serif;
```

**Review by tamalCodes — APPROVED** [review]
# Amazing work 🎉
**The changes look good to me and will be merged soon.**

Do follow [Tamal](https://github.com/tamalCodes) for more Opensource fun projects and don't forget to drop a star so that you get updated about our latest [releases](https://github.com/MilanCommunity/Milan/releases) (we will tag you and mention your work) and also a [shoutout](https://www.linkedin.com/feed/update/urn:li:activity:7073574804519354368/) on social media (LinkedIn and Twitter) !

Happy Opensource 🚀.

**Comment by MahendraDani** [comment]
@tamalCodes Thanks for merging this pull request.
If any issues need addressing I am up for grabs!

**Pull request merged**